In [24]:
import numpy as np
import pandas as pd
import statsmodels.api as sm

# Reading the Data

In [7]:
data = pd.read_excel('Bank_Personal_Loan_Modelling.xlsx' ,sheet_name= 0)

In [8]:
data.head()

ID  Age  Experience  Income  ZIP Code  Family  CCAvg  Education  Mortgage  \
0   1   25           1      49     91107       4    1.6          1         0   
1   2   45          19      34     90089       3    1.5          1         0   
2   3   39          15      11     94720       1    1.0          1         0   
3   4   35           9     100     94112       1    2.7          2         0   
4   5   35           8      45     91330       4    1.0          2         0   

   Personal Loan  Securities Account  CD Account  Online  CreditCard  
0              0                   1           0       0           0  
1              0                   1           0       0           0  
2              0                   0           0       0           0  
3              0                   0           0       0           0  
4              0                   0           0       0           1

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   ID                  5000 non-null   int64  
 1   Age                 5000 non-null   int64  
 2   Experience          5000 non-null   int64  
 3   Income              5000 non-null   int64  
 4   ZIP Code            5000 non-null   int64  
 5   Family              5000 non-null   int64  
 6   CCAvg               5000 non-null   float64
 7   Education           5000 non-null   int64  
 8   Mortgage            5000 non-null   int64  
 9   Personal Loan       5000 non-null   int64  
 10  Securities Account  5000 non-null   int64  
 11  CD Account          5000 non-null   int64  
 12  Online              5000 non-null   int64  
 13  CreditCard          5000 non-null   int64  
dtypes: float64(1), int64(13)
memory usage: 547.0 KB


In [26]:
data.isnull().any()

ID                    False
Age                   False
Experience            False
Income                False
ZIP Code              False
Family                False
CCAvg                 False
Education             False
Mortgage              False
Personal Loan         False
Securities Account    False
CD Account            False
Online                False
CreditCard            False
dtype: bool

In [27]:
data.describe()

ID          Age   Experience       Income      ZIP Code  \
count  5000.000000  5000.000000  5000.000000  5000.000000   5000.000000   
mean   2500.500000    45.338400    20.104600    73.774200  93152.503000   
std    1443.520003    11.463166    11.467954    46.033729   2121.852197   
min       1.000000    23.000000    -3.000000     8.000000   9307.000000   
25%    1250.750000    35.000000    10.000000    39.000000  91911.000000   
50%    2500.500000    45.000000    20.000000    64.000000  93437.000000   
75%    3750.250000    55.000000    30.000000    98.000000  94608.000000   
max    5000.000000    67.000000    43.000000   224.000000  96651.000000   

            Family        CCAvg    Education     Mortgage  Personal Loan  \
count  5000.000000  5000.000000  5000.000000  5000.000000    5000.000000   
mean      2.396400     1.937913     1.881000    56.498800       0.096000   
std       1.147663     1.747666     0.839869   101.713802       0.294621   
min       1.000000     0.000000     1.000000     0.000000       0.000000   
25%       1.000000     0.700000     1.000000     0.000000       0.000000   
50%       2.000000     1.500000     2.000000     0.000000       0.000000   
75%       3.000000     2.500000     3.000000   101.000000       0.000000   
max       4.000000    10.000000     3.000000   635.000000       1.000000   

       Securities Account  CD Account       Online   CreditCard  
count         5000.000000  5000.00000  5000.000000  5000.000000  
mean             0.104400     0.06040     0.596800     0.294000  
std              0.305809     0.23825     0.490589     0.455637  
min              0.000000     0.00000     0.000000     0.000000  
25%              0.000000     0.00000     0.000000     0.000000  
50%              0.000000     0.00000     1.000000     0.000000  
75%              0.000000     0.00000     1.000000     1.000000  
max              1.000000     1.00000     1.000000     1.000000

# Loading X and Y values

In [15]:
y = data['Personal Loan']

In [16]:
x = data[['Age','Experience','Income','CCAvg','Education','Mortgage']]
x

Age  Experience  Income  CCAvg  Education  Mortgage
0      25           1      49    1.6          1         0
1      45          19      34    1.5          1         0
2      39          15      11    1.0          1         0
3      35           9     100    2.7          2         0
4      35           8      45    1.0          2         0
...   ...         ...     ...    ...        ...       ...
4995   29           3      40    1.9          3         0
4996   30           4      15    0.4          1        85
4997   63          39      24    0.3          3         0
4998   65          40      49    0.5          2         0
4999   28           4      83    0.8          1         0

[5000 rows x 6 columns]

In [18]:
x1 = sm.add_constant(x) 

In [21]:
Logistic= sm.Logit(y,x1)

In [22]:
result=Logistic.fit()

Optimization terminated successfully.
         Current function value: 0.157032
         Iterations 9


# Result

In [23]:
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:          Personal Loan   No. Observations:                 5000
Model:                          Logit   Df Residuals:                     4993
Method:                           MLE   Df Model:                            6
Date:                Sat, 08 Aug 2020   Pseudo R-squ.:                  0.5034
Time:                        21:43:10   Log-Likelihood:                -785.16
converged:                       True   LL-Null:                       -1581.0
Covariance Type:            nonrobust   LLR p-value:                     0.000
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -9.7891      1.441     -6.791      0.000     -12.614      -6.964
Age           -0.0736      0.056     -1.324      0.185      -0.183       0.035
Experience     0.0813      0.055      1.466      0.143      -0.027       0.190
Income         0.0492      0.002     22.486      0.000       0.045       0.053
CCAvg          0.1258      0.035      3.619      0.000       0.058       0.194
Education      1.7877      0.104     17.242      0.000       1.585       1.991
Mortgage       0.0008      0.000      1.699      0.089      -0.000       0.002
==============================================================================
"""

Hence we can say that

__P(Loan = Yes/No) = 1 / (1+e^-(-9.7891 - 0.0736(Age) + 0.0813(Experience) + 0.0492(Income) + 0.1258(CCAvg) + 1.7877(Education) + 0.0008(Mortgage))__ 